### PR TITLE
Refactor net-vm

### PR DESCRIPTION
**Pull request by mbssrc**
## Description of changes

Refactoring of netvmExtraModules:
  - remove netvm extra modules file and use pci passthrough module
  - create wifi config service
  - refactor host names to module to allow query ips
  - create reference service module + passthrough to VMs
  - moved dendrite as first reference service

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [ ] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [x] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status

### Testing done 
- Build and run on x1
- checked `smcroute` service is running in net-vm 
- build with reference service disabled

### Testing required
- verify build on all platforms
- build and test on x1
- verify dendrite-pinecone + element works
- verify build works without dendrite-pinecone
- verify build works without reference services

### CODE REVIEW

File: targets/lenovo-x1/everything.nix [hunk 1]
```diff
@@ -79,9 +76,8 @@
                 microvm = {
                   netvm = {
                     enable = true;
-                    extraModules = [
-                      config.ghaf.hardware.passthrough.netvmPCIPassthroughModule
-                    ];
+                    dendrite = true;
```
Comment (brianmcgillion) [inline targets/lenovo-x1/everything.nix:79]: netvm.extraModules = [somewhere.dendrite.module];

File: modules/microvm/virtualization/microvm/modules.nix [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        serverIpAddr = head (map (x: x.ip) getElementVmEntry);
+      in {
+        enable = true;
+        firewallConfig = true;
+        inherit externalNic;
+        inherit internalNic;
+        inherit serverIpAddr;
+      };
+    }
+  ];
```
Comment (brianmcgillion) [inline modules/microvm/virtualization/microvm/modules.nix:43]: as a module that is not here, but under reference-apps/reference-services/ anywhere-else. It is not part of the ghaf framework. It is needed by one application in one specific configuration

File: targets/lenovo-x1/everything.nix (hunk 1 above)
Comment (mbssrc) [reply targets/lenovo-x1/everything.nix]: done

File: modules/microvm/virtualization/microvm/modules.nix (hunk 2 above)
Comment (mbssrc) [reply modules/microvm/virtualization/microvm/modules.nix]: done

File: modules/common/networking/hosts.nix [hunk 3]
```diff
@@ -58,17 +78,55 @@
...
+    };
+  };
+
+  config = mkIf cfg.enable {
+    ghaf.networking.hosts = {
+      inherit entries;
+    };
+
+    # Generate hosts file
+    environment.etc.hosts = lib.mkForce {
```
[the remark this answers is not in the document]
Comment (mbssrc) [reply modules/common/networking/hosts.nix:127]: Yes - the networking definition is being re-worked as we speak and this should be addressed in the next iteration.

**Review by brianmcgillion — APPROVED** [review]

**Comment by milva-unikie** [comment]
Hi! Can you please provide more detailed instructions on how these should be tested?

> - verify dendrite-pinecone + element works
> - verify build works without dendrite-pinecone
> - verify build works without reference services

- How should it be confirmed that dendrite-pinecone + element works?
- How should dendrite-pinecone and reference services be disabled?
- What are the expected changes from disabling them?

**Comment by mbssrc** [comment]
Hey, absolutely, more details below.
Note that the last version is https://github.com/tiiuae/ghaf/commit/ca8ceeb28f404b69e70b3e414a5ee60f054cac14 .

**- How should it be confirmed that dendrite-pinecone + element works?**
Can you test it similar to the original PR (https://github.com/tiiuae/ghaf/pull/533)?
If you cannot test this end-to-end, at least the following should be verified:

- `smcroute.service` is running in the netvm
- `dendrite-pinecone.service` is running in the element-vm

**- How should dendrite-pinecone and reference services be disabled?**
In everything.nix, there is the configuration:
```
reference.services = {
    enable = true;
    dendrite = true;
  };
```
Setting these options to `false` respectively should do the trick. 

**- What are the expected changes from disabling them?**
1. Build on x1 works fine
2. Aforementioned services are not present/running if either of them are turned off

**Comment by johannarautanen** [comment]
Check with commit: ca8ceeb28f404b69e70b3e414a5ee60f054cac14

**NVIDIA Jetson AGX Orin native, booted Orin AGX from eMMC**

- ci-test-automation tests were ok
- all apps launched manually ok
- All manual regression test ok


**NVIDIA Jetson NX Orin native, booted Orin NX from USB flash drive**

- ci-test-automation tests were ok
- all apps launched manually ok
- All manual regression test ok

**Comment by barnabakos** [comment]
with commits up to: ca8ceeb28f404b69e70b3e414a5ee60f054cac14

platform: **Lenovo X1 Carbon**
- **all build variations:**
  - ci-test-automation suite: ok
  - all apps start
  - ad-hoc tesing: no new findings
- **default commit** (both dendrite-pinecone & reference services enabled):
  - smcroute & dendrite services are running; logs are shown
- only **dendrite-pinecone** disabled:
  - smcroute & dendrite services are not running & not available
- only **reference services** disabled:
  - smcroute is not running & not available
  - **dendrite is running; logs are shown**
- both **dendrite-pinecone & reference services** disabled:
  - smcroute & dendrite services are not running & not available

platform: **NVIDIA Jetson AGX Orin with demo apps; cross-compile**
- ci-test-automation suite: ok (SP-4729 - firefox doesn't start -> known issue)
- all apps start
- ad-hoc tesing: no new findings

File: modules/common/networking/hosts.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (brianmcgillion) [reply modules/common/networking/hosts.nix:127]: Being worked on with Tervis

**Pull request merged**